pkg/ifaces/watcher.go

```diff
@@ -45,27 +47,43 @@ func NewWatcher(bufLen int) *Watcher {
 func (w *Watcher) Subscribe(ctx context.Context) (<-chan Event, error) {
 	out := make(chan Event, w.bufLen)
 
-	nsHandles, err := getNetNSHandles()
+	netns, err := getNetNS()
 	if err != nil {
-		go w.sendUpdates(ctx, netns.None(), out)
+		go w.sendUpdates(ctx, "", out)
 	} else {
-		for _, nsh := range nsHandles {
-			nsHandle := nsh
-			go w.sendUpdates(ctx, nsHandle, out)
+		for _, n := range netns {
+			go w.sendUpdates(ctx, n, out)
 		}
 	}
 	// register to get notification when netns is created or deleted and register for link update for new netns
 	w.netnsNotify(ctx, out)
 	return out, nil
 }
 
-func (w *Watcher) sendUpdates(ctx context.Context, netnsHandle netns.NsHandle, out chan Event) {
+func (w *Watcher) sendUpdates(ctx context.Context, ns string, out chan Event) {
+	var netnsHandle netns.NsHandle
+	var err error
 	log := logrus.WithField("component", "ifaces.Watcher")
 	// subscribe for interface events
 	links := make(chan netlink.LinkUpdate)
-	log.WithField("netns", netnsHandle.String()).Debug("linkSubscribe to receive links update")
-	if err := w.linkSubscriberAt(netnsHandle, links, ctx.Done()); err != nil {
-		log.WithError(err).Errorf("can't subscribe to links netns %s", netnsHandle.String())
+	if err = wait.PollUntilContextTimeout(ctx, 10*time.Microsecond, time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
```suggestion
	if err = wait.PollUntilContextTimeout(ctx, 50*time.Microsecond, time.Second, true, func(ctx context.Context) (done bool, err error) {
```
This is taking quite a few tries on my system at 10 usec.  This is your choice, but you might want to try bumping this up to avoid all the retries.